Base/res/html/misc/new-tab.html

```diff
@@ -0,0 +1,94 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8" />
+    <title>New Tab</title>
+
+    <style>
+        #container {
+            display: flex;
+            align-items: center;
+            justify-content: center;
+        }
+        #main {
+            text-align: center;
+            display: block;
+            width: 100%;
+            max-width: 400px;
+        }
+        #image {
+            image-rendering: pixelated;
+        }
+        #input {
+            width: 100%;
+            padding: 5px;
+        }
+        #search-buttons {
+            display: flex;
+            justify-content: space-between;
+        }
+        button:hover {
+            cursor: pointer;
+        }
+    </style>
+<script src="welcome.js"></script>
+</head>
+
+<body>
+
+<div id="container">
+
+<div id="main">
+<br>
+<img src="/res/icons/32x32/app-browser.png" width="64" height="64" id="image"><br><br>
+
+<form>
+    <input type="search" name="q" id="input"><br><br>
+    <div id="search-buttons">
+    <button type="button" onClick="search('bing')">Bing</button>
+    <button type="button" onClick="search('duckduckgo')">DuckDuckGo</button>
+    <button type="button" onClick="search('frogfind')">FrogFind</button>
+    <button type="button" onClick="search('github')">GitHub</button>
+    <button type="button" onClick="search('google')">Google</button>
+    <button type="button" onClick="search('yandex')">Yandex</button>
+    </div>
+</form>
+<br><br>
+<p>Your user agent is: <b><span id="ua"></span></b></p>
+<p>This page loaded in <b><span id="loadtime"></span></b> ms</p>
+</div>
+</div>
+
+<script>
+    queueMicrotask(function() {
+        document.getElementById("loadtime").innerHTML = performance.now();
+    });
+
+    function search(se) {
+        let query = document.getElementById("input").value;
+
+        if (!query) {
+            return;
+        }
+
+        if (se == "bing") {
+            var url = "https://www.bing.com/search?q="+query;
+        } else if(se == "duckduckgo") {
+            var url = "https://duckduckgo.com/?q="+query;
+        } else if(se == "frogfind") {
+            var url = "https://frogfind.com/?q="+query;
+        } else if(se == "github") {
+            var url = "https://github.com/search?q="+query;
+        } else if(se == "google") {
+            var url = "https://google.com/search?q="+query;
+        } else if(se == "yandex") {
+            var url = "https://yandex.com/search/?text="+query;
+        }
+
+        window.location.href = url;
```

Comment:
This properly encodes and as such avoids some unintended edge cases like a query containing `&` 

```suggestion
        if (se == "bing") {
            var url = new URL("https://www.bing.com/search");
            url.searchParams.set("q", query);
        } else if (se == "duckduckgo") {
            var url = new URL("https://duckduckgo.com");
            url.searchParams.set("q", query);
        } else if (se == "frogfind") {
            var url = new URL("https://frogfind.com");
            url.searchParams.set("q", query);
        } else if (se == "github") {
            var url = new URL("https://github.com/search");
            url.searchParams.set("q", query);
        } else if (se == "google") {
            var url = new URL("https://google.com/search");
            url.searchParams.set("q", query);
        } else if (se == "yandex") {
            var url = new URL("https://yandex.com/search");
            url.searchParams.set("text", query);
        }

        window.location.href = url.toString();
```

Reply:
Let's not declare `url` a dozen times, one `let url;` at the top is better.

Could be done even more nicely with a map, something like

```js
const searchEngines = {
    bing: {
        url: "https://www.bing.com/search",
        searchParameter: "q"
    },
    ...
}
```

Reply:
:+1: wanted to limit my yakshedding :^)

Reply:
Scary new territory making a big PR that does not involve fonts! Thanks for the feedback, searchParams is way better!